### PR TITLE
Refactor legacy `avalon` and `pype` to `AYON`

### LINKED ISSUE #9: Remove OpenPype and avalon references
## Description
Make `AYON` and ayon naming as first class citizen for all functionality and metadata naming in repository.

Ideal goal is that `openpype`, `pype` and `avalon` are not mentioned anywhere in codebase. But that probably won't be possible because of backwards compatibility.

## Codebase requirements
- [ ] Instances and containers do use `AYON_INSTANCE_ID` and `AYON_CONTAINER_ID` instead of `AVALON_INSTANCE_ID` and `AVALON_CONTAINER_ID` (constants from ayon-core).
- [ ] If there are metadata stored to scene, we're referring to them as "ayon" metadata and not "openpype", "pype" or "avalon".
- [ ] If there are metadata related to folder, it is not referred to it as "asset" (OpenPype term) - Use correct folder path or folder name based on functionality.
- [ ] If there are metadata related to product, it is not referred to it as "subset" (OpenPype term)
- [ ] If there are metadata related to product type, it is not referred to it as "family", except for publishing (OpenPype term)

And codebase need to change any of that, we must make sure it can be backwards compatible (read older scenes).

Create separate issues if it should be split to multiple PRs, and close the issue if all conditions are met.

### PR DESCRIPTION
**Pull request by BigRoy**
## Changelog Description

Refactor legacy `avalon` and `pype` to `AYON`

## Additional review information

Update all code to refer to `AYON` instead of legacy names.

Fixes #9 

## Testing notes:

1. Harmony launch shoudl work
2. All tools should work
3. Publishing should work

Existing legacy scenes should also still work.

### CODE REVIEW

**Comment by BigRoy** [comment]
There's still mention of "subset" and "Subset Manager" but it's only related to the tool used (the legacy Subset Manager).

Instead of refactoring it in this PR I think that's best tackled when solving #12 and implementing the new publish system - which would make the usage of the Subset Manager obsolete anyway.

File: client/ayon_harmony/api/TB_sceneOpened.js [hunk 1]
```diff
@@ -367,17 +367,17 @@ function start() {
...
+            app.ayonMenu = true;
         }
     }
 
     var menu = null;
-    if (app.avalonMenu == null) {
+    if (app.ayonMenu == null) {
         menu = menuBar.addMenu(System.getenv('AYON_MENU_LABEL'));
     }
     // menu = menuBar.addMenu('Avalon');
```
Comment (iLLiCiTiT) [inline client/ayon_harmony/api/TB_sceneOpened.js:383]:
```suggestion
```

File: client/ayon_harmony/api/js/AyonHarmonyAPI.js [hunk 2]
```diff
@@ -1,22 +1,22 @@
...
 
 /**
  * Get scene metadata from Harmony.
  * @function
  * @return {object} Scene metadata.
  */
-AvalonHarmony.getSceneData = function() {
-    var metadata = scene.metadata('avalon');
+AyonHarmonyAPI.getSceneData = function() {
+    var metadata = scene.metadata('ayon');
```
Comment (kalisp) [inline client/ayon_harmony/api/js/AyonHarmonyAPI.js:19]: This will not work on legacy files.

Comment (BigRoy) [reply client/ayon_harmony/api/js/AyonHarmonyAPI.js:19]: How about this? https://github.com/ynput/ayon-harmony/pull/14/commits/55228cc3a9fdfece00b7163fd95fb2af4b7d2466

**Review by kalisp — APPROVED** [review]
Works with existing instances (still in old Pyblish!)

**Pull request merged**